sdkproject/Assets/Mapbox/Core/Unity/MeshGeneration/Factories/TerrainFactory.cs

```diff
@@ -153,12 +153,22 @@ private void GenerateTerrainMesh(UnityTile tile, float heightMultiplier)
                     var xx = Mathd.Lerp(tile.Rect.Min.x, tile.Rect.Max.x, xrat);
                     var yy = Mathd.Lerp(tile.Rect.Min.y, tile.Rect.Max.y, yrat);
 
+                    var height = heightMultiplier * Conversions.GetRelativeHeightFromColor(tile.HeightData.GetPixel(
+                        (int)(xrat * 255),
+                        (int)((1 - yrat) * 255)),
+                        tile.RelativeScale);
+
+                    /*
+                        Assuming this is in meters? The tallest point on earth is ~8,500m.
+                        Therefore this number should be pretty safe.
+                    */
+                    if (height > 100000 * tile.RelativeScale * heightMultiplier) {
```

Comment:
This will likely work. However, we have determined the missing data is because of bad rendering on our side. Textures with missing data will have width less than 256 (usually 8?). Therefore, it is likely more correct to set `heightMultiplier` to zero in `CreateTerrainHeight()`:

```cs
var texture = new Texture2D(256, 256);
texture.wrapMode = TextureWrapMode.Clamp;
texture.LoadImage(pngRasterTile.Data);
if (texture.width < 256)
{
	heightMultiplier = 0f;
}
tile.HeightData = texture;
tile.HeightDataState = TilePropertyState.Loaded;
GenerateTerrainMesh(tile, heightMultiplier);
```